src/Parser/DemonHunter/Vengeance/Modules/Statistics/SoulFragmentsConsume.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import ExpandableStatisticBox from 'Interface/Others/ExpandableStatisticBox';
+import { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+
+import SPELLS from 'common/SPELLS/index';
+import SpellIcon from 'common/SpellIcon';
+
+import SoulFragmentsTracker from '../Features/SoulFragmentsTracker';
+
+
+class SoulFragmentsConsume extends Analyzer {
+  static dependencies = {
+    soulFragmentsTracker: SoulFragmentsTracker,
+  };
+
+  castTimestamp = 0;
+  totalSoulsConsumed = 0;
+
+  soulsConsumedBySpell = {};
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SPIRIT_BOMB_TALENT.id && spellId !== SPELLS.SOUL_CLEAVE.id && spellId !== SPELLS.SOUL_BARRIER_TALENT.id) {
+      return;
+    }
+    if (!this.soulsConsumedBySpell[spellId]) {
+      this.soulsConsumedBySpell[spellId] = {name: 0};
+      this.soulsConsumedBySpell[spellId] = {souls: 0};
+      this.soulsConsumedBySpell[1000000] = {souls: 0};
+      this.soulsConsumedBySpell[1000000] = {name: 0};
+      this.soulsConsumedBySpell[1000001] = {souls: 0};
+      this.soulsConsumedBySpell[1000001] = {name: 0};
+    }
+    this.soulsConsumedBySpell[spellId].name = event.ability.name;
+    this.castTimestamp = event.timestamp;
+    this.trackedSpell = spellId;
+  }
+
+  on_byPlayer_removebuffstack(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SOUL_FRAGMENT_STACK.id) {
+      return;
+    }
+    if (event.timestamp - this.castTimestamp < 100) {
+      this.soulsConsumedBySpell[this.trackedSpell].souls += 1;
+      this.totalSoulsConsumed += 1;
+      }
+    }
+
+  on_byPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SOUL_FRAGMENT_STACK.id) {
+      return;
+    }
+    if (event.timestamp - this.castTimestamp < 100) {
+      this.soulsConsumedBySpell[this.trackedSpell].souls += 1;
+      this.totalSoulsConsumed += 1;
+    }
+  }
+
+
+  statistic() {
+    this.soulsConsumedBySpell[1000000].souls = this.soulFragmentsTracker.soulsWasted;
+    this.soulsConsumedBySpell[1000000].name = 'Overcap';
+    this.soulsConsumedBySpell[1000001].souls = this.soulFragmentsTracker.soulsGenerated - this.soulFragmentsTracker.currentSouls - this.soulFragmentsTracker.soulsWasted - this.totalSoulsConsumed;
+    this.soulsConsumedBySpell[1000001].name = 'By Touch';
```

Comment:
Looks like you're trying to jam 'Overcap' and 'By Touch' into a map with spells, which is difficult to read and I'm not really on board with it. I see two possible solutions:

1) Set consts like `OVERCAP_ID = 1000000` and `TOUCH_ID = 1000001` and index by those in code so it's easier to understand what's going on.
2) Stop putting these stats in the map, and just add entries directly in the table below.

I prefer option (2) unless there's some technical limitation I'm not thinking of that would make this more annoying

Reply:
I didnt think about doing it that way, since i was pulling all the values from the object i just decided to shove things into it. but yeah i should be able to add it to the table instead.